File: cppDir/tableTest.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
//#include "Deck.h"
#include "HandAlyzer.h"

using namespace std;

void importDeck(const string& fileName, Card hands[][5]);
void printHands(const Card hands[][5]);

int main(int argc, char* argv[]) {
    bool gameMode = true;  // true = Random, false = PreFab Deck
    string fileName = " ";

    if (argc > 1) {
        gameMode = false;
        fileName = argv[1];
    }

    Card hands[6][5];

    cout << "*** P O K E R  H A N D  A N A L Y Z E R ***\n\n\n";

    if (gameMode) {
        cout << "*** USING RANDOMIZED DECK OF CARDS ***\n\n"
             << "*** Shuffled 52 card deck:" << endl;
        Deck deck;
        deck.printDeck();

        for (int playerIndex = 0; playerIndex < 6; playerIndex++){
          for (int i = 0; i < 5; i++) {
              hands[playerIndex][i] = deck.dealCard();
          }
        }
      
        printHands(hands);

        cout << "\n*** Here is what remains in the deck..." << endl;
        deck.printDeck();
    } else {
        importDeck(fileName, hands);
        printHands(hands);
    }

    // Create analyzer to analyze the entire set of hands, then print results
    HandAlyzer analyzer(hands);
    cout << "\n";
    analyzer.getFinalScorePrint();

    return 0;
}

// Function to print hands
void printHands(const Card hands[6][5]) {
    cout << "\n*** Here are the six hands..." << endl;

    for (int i = 0; i < 6; i++) {
        for (int j = 0; j < 5; j++) {
            cout << hands[i][j].toString();
        }
        cout << endl;
    }
}

// Function to import a deck from a file
void importDeck(const string& fileName, Card hands[6][5]) {
    cout << "*** USING TEST DECK ***" << endl;
    cout << "*** File: " << fileName << endl;

    // Vector to store cards after construction
    vector<Card> importedCards;
    // Duplication check variables
    bool duplicateCardBool = false;
    string duplicateCardStr = "";

    ifstream input(fileName);

    string line;

    while (getline(input, line)) {
        cout << line << endl;
        string::size_type pos = 0;
        while (pos < line.length() - 1) {
            string rankParse = line.substr(pos, 2);
            pos += 2;
            string suitParse = line.substr(pos, 1);
            pos += 2;
            
            Card inputCard(rankParse, suitParse); 
            
//            //Debugging tests 
//            cout << rankParse << "_" << suitParse << endl;
//            inputCard.toString(); 
//            //Debugging tests
            
            // Check for duplicates pt1, if found return dupCardBool true
            for (Card checkCard : importedCards) {
                if (checkCard.equals(inputCard)) {
                    duplicateCardBool = true;
                    duplicateCardStr = inputCard.toString();
                }
            }

            // Add card to queue
            importedCards.push_back(inputCard);
        }
    }

    // Check for duplicates pt2
    if (duplicateCardBool) {
        cout << "*** ERROR - DUPLICATED CARD FOUND IN DECK ***" << endl;
        cout << "*** DUPLICATE: " << duplicateCardStr << " ***" << endl;
        exit(1);
    }

    // Deal cards to players from importedCards Queue
    // For each 6 players, fills all 5 slots with a card then goes to the next player
    for (int i = 0; i < 6; i++) {
        for (int j = 0; j < 5; j++) {
            hands[i][j] = importedCards[j + (i * 5)];
        }
    }
}

File: pyDir/tableTest.py
# Willem Ethan Wilcox, CSC 330, Dr. Pounds, Mercer University
from Deck import Card, Deck
from HandAlyzer import HandAlyzer

import sys


# ---------------------------------------- Main()
def main():
    # Initialize game mode and file name
    game_mode = True  # True for Random, False for PreFab Deck
    file_name = ""

    if len(sys.argv) > 1:
       game_mode = False
       file_name = sys.argv[1]

    print("*** P O K E R  H A N D  A N A L Y Z E R ***\n\n")

    # Create 2D list for players' decks
    hands = [[None] * 5 for _ in range(6)]

    # Deal cards based on game mode
    if game_mode:
        # Announce game mode and print deck
        print("*** USING RANDOMIZED DECK OF CARDS ***\n")
        deck = Deck()
        deck.print_deck()

        # Deal 6 hands of 5 cards each
        for i in range(5):
            for j in range(6):
                hands[j][i] = deck.deal_card()

        # Print hands dealt and remaining random deck
        print_hands(hands)
        deck.print_deck()
    else:
        # Deal cards from prefab deck
        hands = import_deck(file_name, hands)

        # Print hands dealt
        print_hands(hands)

    # Create analyzer to analyze entire set of hands, then print results
    analyzer = HandAlyzer(hands)
    analyzer.get_final_score_print()


# ---------------------------------------- Methods
# Print hands function
def print_hands(hands):
    print("*** Here are the six hands...")
    for i in range(6):
        for j in range(5):
            print(hands[i][j], end=" ")
        print()
    print()


# Import deck function
def import_deck(file_name, hands):
    print("*** USING TEST DECK ***\n")
    print("*** File:", file_name)

    # Queue to store cards after construction
    imported_cards = []
    # Duplication check variables
    duplicate_card_bool = False
    duplicate_card_str = ""

    # Read file and split into lines
    with open(file_name, 'r') as file:
        lines = file.readlines()

    for line in lines:
        line = line.strip()
        print(line)
        parts = line.split(',')
        for part in parts:
            last_two_char = part.strip()[-2:]
            if last_two_char[0] == "0":
                rank_parse = "10"
            else:
                rank_parse = last_two_char[0]
            suit_parse = last_two_char[1]
            input_card = Card(rank_parse, suit_parse)
            # Check for duplicates
            if input_card in imported_cards:
                duplicate_card_bool = True
                duplicate_card_str = str(input_card)
            imported_cards.append(input_card)

    # Check for duplicates
    if duplicate_card_bool:
        print("*** ERROR - DUPLICATED CARD FOUND IN DECK ***")
        print("\n*** DUPLICATE:", duplicate_card_str, "***\n")
        sys.exit(0)

    # Deal cards to players from imported_cards list
    for i in range(6):
        for j in range(5):
            hands[i][j] = imported_cards.pop(0)

    return hands


if __name__ == "__main__":
    main()
